qwerty_pyrt/rust/src/wrap_ast.rs

```diff
@@ -390,6 +390,13 @@ impl Expr {
             },
         }
     }
+
+    pub fn typecheck_expr(&self, py: Python<'_>, env: &mut TypeEnv) -> PyResult<Type> {
+        match typecheck_expr(&self.expr, &mut env.check) {
+            Ok(ty) => Ok(Type { ty }),
+            Err(e) => Err(get_err(py, e.kind.to_string(), e.dbg)),
+        }
+    }
```

Comment:
For consistency with `Program::type_check()`:
```suggestion
    pub fn type_check(&self, py: Python<'_>, env: &mut TypeEnv) -> PyResult<Type> {
        typecheck::typecheck_expr(&self.expr, &mut env.check)
            .map(|ty| Type { ty })
            .map_err(|err| get_err(py, err.kind.to_string(), err.dbg))
    }
```
Also maybe you will find `Result::map()` and `Result::map_err()` interesting. I found them on [this page][1]

[1]: https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err